### PR TITLE
[FIXED] Updating consumer filter subject possible panic and deadlock.

### PR DESCRIPTION
**Pull request by derekcollison**
For updating a consumer filter subject make sure locking order is correct and that our sublist is present.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -4688,25 +4688,41 @@ func (mset *stream) removeConsumerAsLeader(o *consumer) {
 }
 
 // swapSigSubs will update signal Subs for a new subject filter.
-// Lock should be held.
+// conumer lock should not be held.
```
Comment (wallyqs) [inline server/stream.go:4691]:
```suggestion
// consumer lock should not be held.
```

Comment (derekcollison) [reply server/stream.go]: fixed.

**Review by Jarema — APPROVED** [review]
LGTM (assuming green CI)

**Comment by derekcollison** [comment]
Will wait on green tests.

**Pull request merged**